src/idl_gen_cpp.cpp

```diff
@@ -161,13 +164,13 @@ class CppGenerator : public BaseGenerator {
 
   void GenIncludeDependencies() {
     int num_includes = 0;
-    for (auto it = parser_.native_included_files_.begin();
-         it != parser_.native_included_files_.end(); ++it) {
+    auto &native_inc_files_ = parser_.native_included_files_;
```

Comment:
Please do not introduce new local variables for such minor level of repetition, this makes code harder to read (since you now have to understand the lifetime of the variable, and what it was defined as). Here and elsewhere.